awx_collection/plugins/modules/role.py

```diff
@@ -261,8 +266,13 @@ def main():
                     lookup_data_populated = {}
                 else:
                     lookup_data_populated = lookup_data
-
-            resource_data.setdefault(key, []).append(module.get_one(key, name_or_id=resource, data=lookup_data_populated))
+            data = module.get_one(key, name_or_id=resource, data=lookup_data_populated)
+            if data is None:
+                module.fail_json(
+                    msg='Unable to find {0} with name: {1}'.format(key, resource), changed=False
```

Comment:
This comment applies for both loops here.

In the case where >1 item was missing this would just give you the last message for an item that it failed to find. Can we make an array like `missing_items = []` and append these messages to the array. After the loops, before the module changes things we could exit early in this scenario to prevent a partial update with something like:
```
if len(missing_items) > 0:
    msg='There were {} missing items, see missing_items for more details'.format(len(missing_items))
    module.fail_json(msg, missing_items, changed=False)
```

(I don't remember if ^ is that exact syntax for exiting with another parameter or not)

Reply:
In practice its usually one item, but I can see about doing it in loop to catch more.

Reply:
Did testing around this, for the user/team keys it fails before you get to the apply to list so I am leaving it as my ofiginal change, as there is only one item provided from the user, and it fails if the thing to be granted permissions to act on lists of objects does not exist.